src/barril/units/_quantity.py

```diff
@@ -3,28 +3,67 @@
 """
 
 from collections import OrderedDict
-from typing import List
+from typing import (
+    List,
+    Optional,
+    ClassVar,
+    Dict,
+    Union,
+    Any,
+    Tuple,
+    Sequence,
+    overload,
+    NoReturn,
+    TypeVar,
+)
 
 from barril.units.exceptions import QuantityValidationError
-from barril.units.unit_database import InvalidUnitError, UnitDatabase, UnitsError, FixUnitIfIsLegacy
+from barril.units.unit_database import (
+    InvalidUnitError,
+    UnitDatabase,
+    UnitsError,
+    FixUnitIfIsLegacy,
+    CategoryInfo,
+)
 from oop_ext.interface import ImplementsInterface
 
 from ._unit_constants import UNKNOWN_UNIT
-from .interfaces import IQuantity, IQuantity2, IQuantity3, IQuantity6
+from .interfaces import IQuantity, IQuantity2, IQuantity3, IQuantity6, UnitExponentTuple
 
 __all__ = ["Quantity", "ObtainQuantity"]
 
 
-def _ObtainReduced(state):
-    """
-    :param list state:
-        The value returned from _Quantity.__reduce__ (used in pickle protocol).
-    """
-    unknown_unit_caption = state.pop(-1)
-    return ObtainQuantity(OrderedDict(state), None, unknown_unit_caption)
+@overload
+def ObtainQuantity(
+    unit: str,
+    category: Optional[str] = None,
+    unknown_unit_caption: Optional[str] = None,
+) -> "Quantity":
+    ...
+
+
+@overload
+def ObtainQuantity(
+    unit: str,
+    category: Union[Tuple[str, ...], str],
+) -> "Quantity":
+    ...
 
 
-def ObtainQuantity(unit, category=None, unknown_unit_caption=None):
+@overload
+def ObtainQuantity(
+    unit: Union[None, List[UnitExponentTuple], Dict[str, Sequence[Union[str, int]]]],
+    category: Optional[str] = None,
+    unknown_unit_caption: Optional[str] = None,
+) -> "Quantity":
+    ...
+
+
+def ObtainQuantity(
+    unit: Union[str, None, List[UnitExponentTuple], Dict[str, Sequence[Union[str, int]]]],
```

Comment:
👀

Reply:
Yep... that goes to show how type annotations actually *expose* hidden complexity.